packages/tupaia-web-server/src/utils/generateFrontendExcludedFilter.ts

```diff
@@ -3,30 +3,93 @@
  * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
 
-interface FrontEndExcludedConfig {
-  types: string[];
-}
+import { AccessPolicy } from '@tupaia/access-policy';
+import { PermissionGroup } from '@tupaia/types';
+import { TupaiaWebServerModelRegistry } from '../types';
+
+type FrontEndExcludedException = {
+  permissionGroups: PermissionGroup['name'][];
+};
+
+const getShouldExcludeTypes = async (
+  models: TupaiaWebServerModelRegistry,
+  exceptions?: FrontEndExcludedException,
+  accessPolicy?: AccessPolicy,
+) => {
+  // If there are no exceptions, then exclude the types
+  if (!exceptions) return true;
+  const allCountries = await models.country.find({});
+
+  const allCountryCodes = allCountries.map(country => country.code);
+
+  // If there are no permission groups in the exceptions, then exclude the types
+  if (!exceptions?.permissionGroups) {
+    return true;
+  }
+
+  const permissionGroups = exceptions?.permissionGroups || [];
+  const userPermissionGroups = accessPolicy
+    ? accessPolicy?.getPermissionGroups(allCountryCodes)
+    : [];
+
+  const userHasAccessToExcludedTypes = permissionGroups.some(permissionGroup =>
+    userPermissionGroups.includes(permissionGroup),
+  );
+
+  // If the user has any of the permission groups in the exceptions, then do not exclude the types
+  return !userHasAccessToExcludedTypes;
+};
+
+export const getTypesToExclude = async (
+  models: TupaiaWebServerModelRegistry,
+  accessPolicy: AccessPolicy | undefined,
+  projectCode: string,
+  useDefaultIfNoExclusions = true,
+) => {
+  const project = await models.project.find({
+    code: projectCode,
+  });
+
+  if (!project[0]) throw new Error(`Project with code '${projectCode}' not found`);
+
+  const { config } = project[0];
+  const { typesExcludedFromWebFrontend } = models.entity;
```

Comment:
We should be able to delete this functionality, in favour of configuring all the relevant projects to have `case` and `case_contact` in their front-end excluded types, right? Sorry to ask for a bit more work here, just wanna simplify things where possible